tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/Lambda/LambdaCommon.cs

```diff
@@ -25,10 +25,15 @@ internal abstract class LambdaCommon
 
     internal static Scope CreatePlaceholderScope(Tracer tracer, NameValueHeadersCollection headers)
     {
-        var spanContext = SpanContextPropagator.Instance.Extract(headers);
-        TelemetryFactory.Metrics.RecordCountSpanCreated(MetricTags.IntegrationName.AwsLambda);
-
-        var span = spanContext != null ? tracer.StartSpan(PlaceholderOperationName, tags: null, parent: spanContext, serviceName: PlaceholderServiceName, addToTraceContext: false) : tracer.StartSpan(PlaceholderOperationName, tags: null, serviceName: PlaceholderServiceName, addToTraceContext: false);
+        var context = SpanContextPropagator.Instance.Extract(headers);
```

Comment:
I'm noticing in a bunch of places that you have `SpanContextPropagator.Instance.Extract` followed by `Baggage.Current.Merge`, which introduces an opportunity for us to miss the inclusion in all the places where it should be added. Can we introduce an overload like `ExtractAndSetBaggage` so we don't miss this? Or even a method that chains like `SpanContextPropagator.Instance.Extract(headers).MergeWithCurrentBaggage();` that returns itself could suffice. What do you think?

Reply:
I had the same concern 😅. I thought about merging the baggage inside of `Extract()` so we never forget to do it, but I wanted to keep the extraction operation separate from what we do later with the extracted objects, `SpanContext` and `Baggage`, since they are independent of each other, and merging the extracted baggage into `Baggage.Curret` modifies static state.

Having an extension method like `MergeWithCurrentBaggage` does seem useful, but I'm not sure that will help with the issue of us missing it accidentally. Unless we can write an analyzer for that?

I not super happy with this as it is now so open to suggestions!

Reply:
Maybe we can add an overload `SpanContextPropagator.Instance.ExtractToCurrent` that does this? That way we can track usages of `SpanContextPropagator.Instance.Extract` vs `SpanContextPropagator.Instance.ExtractToCurrent`

Reply:
🧠 🌩️  How about an "overload" that takes a baggage instance? If provided, it merges the extracted baggage into the specified baggage. That way the _caller_ is the one accessing static state instead of `SpanContextPropagator`. Something like:

```csharp
PropagationContext ExtractContextAndMergeBaggage(..., Baggage? baggage)
{
    var context = Extract(...);

    if (baggage is not null)
    {
        baggage.Merge(context.Baggage);
    }
}


// usage
var context = SpanContextPropagator.Instance.Extract(headers, Baggage.Current);
```

Reply:
I do like that `Baggage.Current` is accessed from the caller, but the `Merge` operation here feels different than the `Baggage.Merge` operation you've introduced. In those calls, we merge the argument `Baggage` into `this`, but in this new ExtractContextAndMergeBaggage method we're merging the extracted `Baggage` into the argument `Baggage`. Maybe all it needs is a rename to ExtractContextAndMergeIntoBaggage?

Reply:
I'm having another issue here 😓 . We have multiple overloads of `Extract`, so I would need to add the new `ExtractContextAndMergeIntoBaggage` for each one:

```csharp
PropagationContext Extract<TCarrier>(TCarrier headers)
PropagationContext Extract<TCarrier>(TCarrier carrier, Func<TCarrier, string, IEnumerable<string?>> getter)
PropagationContext Extract<TCarrier, TCarrierGetter>(TCarrier carrier, TCarrierGetter carrierGetter)
PropagationContext Extract(IReadOnlyDictionary<string, string?>? serializedSpanContext)
```

We could avoid the 4 `ExtractAndMerge` overloads by adding this "merge to current baggage" action as an extension method to `PropagationContext` so it can be combined with any of the four `Extract` overloads:

```csharp
var context = SpanContextPropagator.Instance.Extract(...).MergeBaggageInto(Baggage.Current);
```

-----

> but the `Merge` operation here feels different than the `Baggage.Merge` operation you've introduced

Agreed. I'll think about renaming them to `MergeInto`/`MergeFrom` or something like that to make it clear in which direction we're merging data.

Reply:
As discussed via chat, we're going with a single extension method on `PropagationContext`.

Reply:
I replaced all of these 
```
var context = SpanContextPropagator.Instance.Extract(...);
Baggage.Current.Merge(context.Baggage);
```
with
```
var context = SpanContextPropagator.Instance.Extract(...).MergeBaggageInto(Baggage.Current);
```